src/mappings/handlers/multisigEventHandler.ts

```diff
@@ -0,0 +1,157 @@
+import { AnyTuple, CallBase } from "@polkadot/types/types";
+
+const callWalk = CreateCallWalk();
+
+export async function handleMultisigCallEvent(event: SubstrateEvent) {
+  if (!event.extrinsic) throw new Error("Extrinsic not found");
+
+  const isNewMultisig = event.event.method === "NewMultisig";
+
+  const possibleIndex = isNewMultisig ? 2 : 3;
+  const callHash = getDataFromEvent<Uint8Array>(event.event, "callHash", possibleIndex);
+
+  if (!callHash) throw new Error("Call hash not found");
+
+  const callHashString = u8aToHex(callHash);
+
+  const multisigPossibleIndex = isNewMultisig ? 1 : 2;
+  const multisig = getDataFromEvent<AccountId>(event.event, "multisig", multisigPossibleIndex);
+
+  if (!multisig) throw new Error("Multisig not found");
+
+  const multisigAccountId = multisig.toHex();
+
+  let timepoint = null;
+  let blockCreated = getBlockCreated(event.extrinsic);
+  let indexCreated = getIndexCreated(event.extrinsic);
+  if (!isNewMultisig) {
+    timepoint = getDataFromEvent<Timepoint>(event.event, "timepoint", 1) || getDataFromEvent<Timepoint>(event.event, "Timepoint", 1);
+  }
+  if (timepoint && timepoint.height && timepoint.index) {
+    blockCreated = timepoint.height.toNumber();
+    indexCreated = timepoint.index.toNumber();
+  }
+
+  const existingOperation = await findExistingOperation(callHashString, blockCreated, indexCreated, multisigAccountId);
+
+  const operationId = generateOperationId(callHashString, multisigAccountId, blockCreated, indexCreated);
+
+  const newOperation = await MultisigOperation.create({
+    ...existingOperation,
+    id: operationId,
+    callHash: callHashString,
+    status: existingOperation?.status || OperationStatus.pending,
+    accountId: multisigAccountId,
+    depositor: u8aToHex(decodeAddress(event.extrinsic.extrinsic.signer.toString())),
+    blockCreated: blockCreated,
+    indexCreated: indexCreated,
+    timestamp: timestamp(event.extrinsic.block),
+  });
+
+  await newOperation.save();
+
+  const handleCall = async (visitedCall: VisitedCall) => {
+    const call = getDataFromCall<CallBase<AnyTuple>>(visitedCall.call, "call");
+
+    const existingOperation = await findExistingOperation(callHashString, blockCreated, indexCreated, multisigAccountId);
+
+    if (!existingOperation) throw new Error("Operation not found and that is fucking bad");
+
+    let callData = call?.toHex() || existingOperation?.callData;
+    let method = call?.method || existingOperation?.method;
+    let section = call?.section || existingOperation?.section;
+
+    const updatedOperation = await MultisigOperation.create({
+      ...existingOperation,
+      callData: callData,
+      method: method,
+      section: section,
+    });
+    await updatedOperation.save();
+
+    const signer = u8aToHex(decodeAddress(visitedCall.origin));
+
+    await MultisigEvent.create({
+      id: generateEventId(updatedOperation.id, signer, EventStatus.approve),
```

Comment:
We launch this handler also on `cancelAsMulti` so always setting `EventStatus.approve` looks incorrect